src/Middleware/StaticFiles/src/StaticFileContext.cs

```diff
@@ -78,56 +76,54 @@ public StaticFileContext(HttpContext context, StaticFileOptions options, PathStr
             _ifModifiedSinceState = PreconditionState.Unspecified;
             _ifUnmodifiedSinceState = PreconditionState.Unspecified;
             _range = null;
-            _isRangeRequest = false;
+            _requestType = RequestType.Unspecified;
         }
 
-        internal enum PreconditionState
-        {
-            Unspecified,
-            NotModified,
-            ShouldProcess,
-            PreconditionFailed
-        }
+        private RequestHeaders RequestHeaders => (_requestHeaders ??= _request.GetTypedHeaders());
 
-        public bool IsHeadMethod
-        {
-            get { return _isHead; }
-        }
+        private ResponseHeaders ResponseHeaders => (_responseHeaders ??= _response.GetTypedHeaders());
 
-        public bool IsRangeRequest
-        {
-            get { return _isRangeRequest; }
-        }
+        public bool IsHeadMethod => (_requestType & RequestType.IsHead) != 0;
 
-        public string SubPath
-        {
-            get { return _subPath.Value; }
-        }
+        public bool IsGetMethod => (_requestType & RequestType.IsGet) != 0;
 
-        public string PhysicalPath
-        {
-            get { return _fileInfo?.PhysicalPath; }
-        }
+        public bool IsRangeRequest => (_requestType & RequestType.IsRange) != 0;
 
-        public bool ValidateNoEndpoint()
-        {
-            // Return true because we only want to run if there is no endpoint.
-            return _context.GetEndpoint() == null;
-        }
+        public string SubPath => _subPath.Value;
+
+        public string PhysicalPath => _fileInfo?.PhysicalPath;
+
+        // Return true because we only want to run if there is no endpoint.
+        public bool ValidateNoEndpoint() => _context.GetEndpoint() == null;
 
         public bool ValidateMethod()
         {
             _method = _request.Method;
-            _isGet = HttpMethods.IsGet(_method);
-            _isHead = HttpMethods.IsHead(_method);
-            return _isGet || _isHead;
+            var isValid = false;
+            if (HttpMethods.IsGet(_method))
+            {
+                _requestType |= RequestType.IsGet;
+                isValid = true;
+            }
+            else
+            {
+                _requestType &= ~RequestType.IsGet;
```

Comment:
Why would you need to clear this? It should have started clear.

Reply:
Was just following the original logic. Will change; also if GET it can't be HEAD